torch/csrc/profiler/collection.cpp

```diff
@@ -149,6 +156,169 @@ void InputOutputEncoder::clear() {
+
+  if (config_.state == ProfilerState::KINETO_GPU_FALLBACK) {
+    try {
+      out->fallback_ = torch_ops_.gpu_fallback_.emplace_back();
+      torch::profiler::impl::cudaStubs()->record(
+          nullptr, &out->fallback_->cuda_event_start_, nullptr);
+    } catch (const std::exception& e) {
+      LOG(WARNING) << "Failed to record CUDA event. " << e.what();
+    }
+  }
+
+  event->start_time_ = torch::profiler::impl::getApproximateTime();
+  event->allow_tf32_cublas_ = at::globalContext().allowTF32CuBLAS();
+  return out;
+}
+
+// ---------------
+// |  Collation  |
+// ---------------
+namespace {
+template <typename T>
+struct StealOrDefault {
+  StealOrDefault(T& container)
+      : container_{container}, it_{container.begin()} {}
+
+  ~StealOrDefault() {
+    container_.get().clear();
+  }
+
+  typename T::Iterator::value_type operator()() {
+    if (it_.exhausted()) {
+      return typename T::Iterator::value_type();
+    } else {
+      auto result = std::move(*it_);
+      ++it_;
+      return result;
+    }
+  }
+
+  std::reference_wrapper<T> container_;
+  typename T::Iterator it_;
+};
+} // namespace
+
+void ThreadLocalSubqueue::TorchOpStorage::materialize(
+    std::vector<std::shared_ptr<Result>>& out,
+    const std::function<time_t(approx_time_t)> time_converter,
+    const uint64_t tid,
+    const kineto::DeviceAndResource& kineto_info) {
+  // Plumb Autograd info to the top level annotation.
+  auto it = op_events_.begin();
+  for (C10_UNUSED const auto _ :
+       c10::irange(static_cast<int64_t>(op_events_.size()) - 1)) {
+    auto& first = it->basic_fields_;
+    auto& second = (++it)->basic_fields_;
+    if (first.scope_ == at::RecordScope::FUNCTION &&
+        second.scope_ == at::RecordScope::BACKWARD_FUNCTION &&
+        first.name_.rfind("autograd::engine::evaluate_function: ", 0) == 0) {
+      first.sequence_number_ = second.sequence_number_;
+      first.forward_tid_ = second.forward_tid_;
+    }
+  }
+
+  auto input_getter = inputs_outputs_.getNextShapesAndDtypes();
+
+  // TODO: CTAD will take care of template args when we move to C++17
+  auto jit_stack = StealOrDefault<decltype(jit_stack_)>(jit_stack_);
+  auto jit_module = StealOrDefault<decltype(jit_modules_)>(jit_modules_);
+  auto extra_args = StealOrDefault<decltype(extra_args_)>(extra_args_);
+  auto gpu_fallback = StealOrDefault<decltype(gpu_fallback_)>(gpu_fallback_);
+
+  for (auto event = op_events_.begin(); event != op_events_.end(); ++event) {
+    ExtraFields<EventType::TorchOp> e{
+        std::move(event->basic_fields_),
+        ThreadLocalSubqueue::TorchOpStorage::OpList::correlationID(event),
+        time_converter(event->end_time_),
+        input_getter(),
+        jit_stack(),
+        jit_module(),
+        extra_args(),
+        gpu_fallback(),
+        event->allow_tf32_cublas_};
+
+    out.emplace_back(Result::create(
+        time_converter(event->start_time_), tid, kineto_info, std::move(e)));
```

Comment:
Hi, @robieta 
Here i find the _**time_converter**_ is always modifying the start time value for the torch ops.